entities/entity.go

```diff
@@ -215,7 +215,9 @@ func (e *Entity) SetMetadata(k, v string) {
 
 // Metadata accesses the value, and whether it existed, for a given metadata key
 func (e *Entity) Metadata(k string) (v string, ok bool) {
-	v, ok = e.metadata[k]
+	if e.metadata != nil {
```

Comment:
This check doesn't accomplish anything in Go, since nil maps permit reads (it has the same behavior as reading a non-existent key on a non-nil map).

As such, this isn't code hardening.